File: original_scripts/c++/is_palindrome/qwen2.5-coder_3b.cpp
```cpp
bool is_palindrome(const std::string& s) {
    for (size_t i = 0; i < s.size() / 2; ++i) {
        if (s[i] != s[s.size() - 1 - i]) {
            return false;
        }
    }
    return true;
}
```

File: original_scripts/r/is_palindrome/qwen2.5-coder_3b.r
```r
is_palindrome <- function(x) {
  str_x <- as.character(x)
  n <- length(str_x)
  for (i in 1:(n/2)) {
    if (str_x[i] != str_x[n-i+1]) return(FALSE)
  }
  TRUE
}
```